pkg/fs/local_file_system_test.go

```diff
@@ -0,0 +1,173 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package fs (file system) is an independent component to operate file and directory.
+package fs
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+var _ = ginkgo.Describe("Loacl File System", func() {
+	const (
+		data          string = "BanyanDB"
+		dirName       string = "dir"
+		fileName      string = "dir/file"
+		flushFileName string = "dir/flushFile"
+	)
+
+	var fs FileSystem
+
+	ginkgo.Context("Local File", func() {
+		ginkgo.BeforeEach(func() {
+			fs = NewLocalFileSystem()
```

Comment:
To prevent cluttering in the current directory, kindly execute the test within the `tmp` folder.